### PR TITLE
Fix `toolbarViewer`/`toolbarContainer` regressions (PR 18385, 18786 follow-up)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
- **Use the "correct" toolbar container element in `getViewerConfiguration` (PR 18385 follow-up)**

   With the changes made in PR #18385 the `toolbarViewer` element is now shorter than before, since the padding is applied on its `toolbarContainer` parent-element instead.
   This causes two separate regressions:
    - Clicking at the very top/bottom of the toolbar no longer closes the secondaryToolbar like previously.
    - The `CaretBrowsingMode`-constructor no longer computes the toolbar-height correctly.

   Given how/where the `container`-property is being used these changes *should* thus be safe.

 - **Update the `CaretBrowsingMode` toolbar-height if the `toolbarDensity` preference changes (PR 18786 follow-up)**

   Otherwise the isVisible-calculations may not work correctly.

### CODE REVIEW

File: web/caret_browsing.js [hunk 1]
```diff
@@ -23,10 +23,20 @@ class CaretBrowsingMode {
...
-    this.#toolBarHeight = toolbarContainer?.getBoundingClientRect().height ?? 0;
+
+    const computeToolBarHeight = () => {
+      this.#toolBarHeight =
+        toolbarContainer?.getBoundingClientRect().height ?? 0;
+    };
+    computeToolBarHeight();
+
+    eventBus._on("toolbardensity", async () => {
+      await Promise.resolve(); // Ensure that the UI has time to update first.
```
Comment (calixteman) [inline web/caret_browsing.js:37]: Is it enough ?
Maybe `setTimeout(..., 0)` will give us slightly more time. 
Normally, the user is changing the toolbar height in using toolbar config ui or from about:config, in both cases `setTimeout` should be enough.
Maybe I'm a bit overthinking here.

Comment (Snuffleupagus) [reply web/caret_browsing.js:37]: Mainly I just wanted to ensure that https://github.com/mozilla/pdf.js/blob/3cdc3258d94a06af658e50290bf709426adf44b8/web/toolbar.js#L129-L140 has run, but I suppose that using `setTimeout` is indeed safer here.

Comment (calixteman) [reply web/caret_browsing.js]: An other possibility would be to add a ResizeObserver on the toolbar.

Comment (Snuffleupagus) [reply web/caret_browsing.js]: Sure, please see the updated patch.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**